### ISSUE TITLE
_mtga_fix="true" is broken on 5.7+

### ISSUE THREAD

**Issue by Nocifer** [Frogging-Family/wine-tkg-git#26]
Hello,

I tried to build Wine 5.7 with _mtga_fix="true" and I got the following error:
```
  -> Applying mtga.patch
patching file dlls/ntdll/thread.c
Hunk #1 FAILED at 778.
Hunk #2 FAILED at 793.
Hunk #3 FAILED at 813.
3 out of 3 hunks FAILED -- saving rejects to file dlls/ntdll/thread.c.rej
==> ERROR: A failure occurred in prepare().
    Aborting...
```
Checking the related bug report at https://bugs.winehq.org/show_bug.cgi?id=45546, there is a new comment by a Wine dev stating that the problematic behaviour that prompted this patch is now supposedly more or less fixed; but building Wine without the patch and trying to download and apply a game update still failed with the exact same error as previously reported (GetThreadContext failed)

So, I guess this needs to be updated for Wine 5.7+.

**Comment by Tk-Glitch** [comment]
Could you please try with the following in a .mypatch and dropped in `/wine-tkg-git/wine-tkg-userpatches`?
```
diff --git a/server/thread.c b/server/thread.c
index cfdf42c756..7c1c1108eb 100644
--- a/server/thread.c
+++ b/server/thread.c
@@ -1848,7 +1848,7 @@ DECL_HANDLER(get_thread_context)
             {
                 assert( reply->self );
                 memset( context, 0, sizeof(context_t) );
-                context->cpu = thread_context->regs.cpu;
+                context->cpu = thread->process->cpu;
                 if (req->flags & system_flags)
                 {
                     get_thread_context( thread, context, req->flags & system_flags );
```

This is currently a pending change in the ML: https://www.winehq.org/pipermail/wine-devel/2020-April/165251.html
If that indeed fixes it I'll make our current MTGA patch legacy.

**Comment by Nocifer** [comment]
Nope, unfortunately it's still happening even with this bug fix applied. This time though it managed to download ~1.2GB of data instead of the ~650MB it downloaded the first time; don't know if that could be of any significance.

Also, I visited the bug thread at https://bugs.winehq.org/show_bug.cgi?id=49011 (linked in that mailing list thread) and when I checked the attached fix, I noticed there was a difference between it and the patch you posted. Instead of
```
-                context->cpu = thread_context->regs.cpu;
+                context->cpu = thread->process->cpu;
```
it's
```
-                context->cpu = thread_context->regs.cpu;
+                context->cpu = current->process->cpu;
```
I tried that too and it still didn't work properly, but it managed to reach almost 2.5GB of data before it crashed. Again, I don't know if this difference in downloaded data is significant or if it's totally random.

Bottom line is, the upstream fixes have not really fixed the issue.
